src/components/datasets/DatasetHeaderButtons/DatasetHeaderButtons.tsx

```diff
@@ -0,0 +1,28 @@
+import { Button, HStack } from "@chakra-ui/react";
+import { BiImport } from "react-icons/bi";
+import { BsStars } from "react-icons/bs";
+
+import { GenerateDataModal } from "./GenerateDataModal";
+import { useState } from "react";
+
+export const DatasetHeaderButtons = () => {
+  const [generateModalOpen, setGenerateModalOpen] = useState(false);
```

Comment:
nit: `useDisclosure` is a thin wrapper around this pattern

Reply:
Okay, let's use it